### ISSUE TITLE
Network install (PXE) reboot indefinitely 

### ISSUE THREAD

**Issue by GannonTdW** [Telmate/terraform-provider-proxmox#685]
Hello,

Step to reproduce:
- create a main.tf with a PXE resource as described in [the documentation](https://registry.terraform.io/providers/Telmate/proxmox/latest/docs/resources/vm_qemu#provision-through-pxe-network-boot):
```terraform
resource "proxmox_vm_qemu" "pxe-minimal-example" {
    name                      = "pxe-minimal-example"
    agent                     = 0
    boot                      = "order=net0;scsi0"
    pxe                       = true
    target_node               = "test"
    network {
        bridge    = "vmbr0"
        firewall  = false
        link_down = false
        model     = "e1000"
    }
}
```
- `terraform apply`
- the resource is created, boots on the network, builds and reboots ... but like `boot = "order=net0;scsi0"`, the VM still boots on the network and restarts a build cycle and reboots etc

I have create a VM "by hand", with the boot order `scsi0,net0` -> No issue. The boot on the  empty scsi0, then on the network, builds, reboot on scsi0.

I tried to use `boot = "order=scsi0;net0"`, but I get the following error

>  Error: no network boot option matched in 'boot' config

Versions:
- Proxmox: 7.3-4
- terraform: v1.3.7 (linux)
- terraform-provider-proxmox: 2.9.11
- VM OS: Fedora CoreOS 37.20221225.3.0

**Comment by tnyeanderson** [comment]
This should be fixed in 2.9.13 by #668 

Could you upgrade and try again?

**Comment by GannonTdW** [comment]
I'm not sure if this is related to your changes but now I have this error

> proxmox_vm_qemu.pxe-minimal-example: Creating...
╷
│ Error: cloud-init parameters only supported on clones or updates
│
│   with proxmox_vm_qemu.pxe-minimal-example,
│   on main.tf line 22, in resource "proxmox_vm_qemu" "pxe-minimal-example":
│   22: resource "proxmox_vm_qemu" "pxe-minimal-example" {
│
╵
The main.tf part:
```
resource "proxmox_vm_qemu" "pxe-minimal-example" {
    name                      = "pxe-minimal-example"
    agent                     = 0
    boot                      = "order=net0;scsi0"
    pxe                       = true
    target_node = var.target_node
    network {
        bridge    = "vmbr0"
        firewall  = false
        link_down = false
        model     = "e1000"
    }
}
```

**Comment by tnyeanderson** [comment]
Not related to my changes, but I should have that cloud-init issue fixed in #696 and https://github.com/Telmate/proxmox-api-go/pull/238 :)

**Comment by GannonTdW** [comment]
Well, my first problem is resolv by the PR https://github.com/Telmate/terraform-provider-proxmox/pull/668
I have new problems that should be fixed with PR https://github.com/Telmate/terraform-provider-proxmox/pull/696  and https://github.com/Telmate/proxmox-api-go/pull/238.
I have opened a new PR for documentation: https://github.com/Telmate/terraform-provider-proxmox/pull/697

I close this one.
